### ISSUE TITLE
Multiple react-s3-uploaders on a page don't call their own uploadFile function

### ISSUE THREAD

**Issue by tbohlen** [odysseyscience/react-s3-uploader#144]
Hey there, been using react-s3-uploader for a little while in a project. I just went to put multiple uploaders on a page, and ran into this issue.

No matter which uploader the user selects a file with, the uploadFile function of the _first_ uploader is always called when the upload starts. That means that my preprocess, onProgress, etc. functions from that first uploader get called for all file uploads on the page, and my handler functions for the other uploaders never get called.

I'm checking with uploadFile function is being called by setting a breakpoint in it, and then checking this.props in the console. I've attached a unique data field to each uploader.

Any idea what is going on & can anyone else reproduce this?

**Comment by seanadkinson** [comment]
Hi @tbohlen,  I haven't seen this issue before unfortunately, and scanning the code, I can't see any reason why this would be the case.  If you can put together a test case that reproduces, I'd be happy to see if I can spot what's going on.

Best, Sean

**Comment by tbohlen** [comment]
yeah, I couldn't see anything in the code that could cause this either...I haven't had a chance to go back and make a small repro case, but will post here if/when I do. For now, I've worked around it by storing some extra state outside the component.

Thanks for the response Sean.

**Comment by tbohlen** [comment]
This was entirely my own fault.... nothing to do with react-s3-uploader. Solved and closing.
